### ISSUE TITLE
Integrate  with aws cognito/amplify

### ISSUE THREAD

**Issue by sankar9659** [invertase/react-native-apple-authentication#109]
Hi,
I'm implementing apple authentication for my app, I have used aws cognito for facebook and other authentication.
So I want to integrate with aws cognito for apple authentication.
Is there any way to integrate it.
or any sample code is available?
Thanks

**Comment by mikehardy** [comment]
Hi there @sankar9659 I have to say I have no idea, this is kind of a "DIY" (Do It Yourself) repo, it provides access to the underlying APIs in javascript, but the integrations are things you have to do.

Here is an example, you'll have to take the idea and bend it to AWS, likely in some sort of separate react-native AWS wrapper that depends on this: https://github.com/invertase/react-native-firebase/pull/2979/files

In particular notice that as developers we use this library to get the token and nonce and send them in https://github.com/invertase/react-native-firebase/pull/2979/files#diff-8cf0dc69a857a1b03cec1bb76724704bR882 - the coupling is very loose

So unless there is a change you need in this library to support AWS like #52 did for Auth0, then there's no action for this library. And if there is something like #52 post a PR that makes the AWS-compatibility change and we can review

**Comment by mkrn** [comment]
For anyone looking, I've got it working like this:
- Use this package, and decode the token to get the email and name
- Sign in with amplify and authenticationFlowType: "CUSTOM_AUTH",
- You will have to implement 3 lambdas to define auth challenge, create auth challenge, and verify auth challenge response
- Add them as hooks to your Cognito user pool
- If your sign-in errors out with UserNotFoundException then sign Up with randomized password
- If your sign in receives challengeName CUSTOM_CHALLENGE respond with the idToken you've got from apple
- In your verify challenge response hook use `verifyAppleToken` npm package to validate the token
- In your define auth hook issue the tokens, voila = you're signed in!

Unfortunately, it's NOT possible to use this with federatedSignIn of amplify (it uses Cognito auth backend to generate cognito users). I've tried many workarounds, with no luck.
